src/__test__/extra/write-after-end.test.ts

```diff
@@ -0,0 +1,42 @@
+import { createTestNode, delay, jsonTestEvents } from "../utils";
+
+import { EventStoreDBClient } from "../..";
+import { UnavailableError } from "../../utils";
+
+describe("write after end", () => {
+  const node = createTestNode();
+  let client!: EventStoreDBClient;
+
+  beforeAll(async () => {
+    await node.up();
+    client = new EventStoreDBClient(
+      { endpoint: node.uri },
+      { rootCertificate: node.rootCertificate },
+      { username: "admin", password: "changeit" }
+    );
+  });
+
+  afterAll(async () => {
+    await node.down();
+  });
+
+  test("Should not write after end", async (done) => {
+    const STREAM_NAME = "json_stream_name";
+    await client.appendToStream(STREAM_NAME, jsonTestEvents());
+
+    client
+      .appendToStream(STREAM_NAME, jsonTestEvents(100_000))
+      .then((result) => {
+        expect(result).toBe("unreachable");
+      })
+      .catch((e) => {
+        expect(e).toBeInstanceOf(UnavailableError);
+      });
+
+    node.killNode(node.endpoints[0]);
+
+    // wait for any unhandled rejections
+    await delay(5_000);
+    done();
```

Comment:
**CONCERN:** I might be wrong, but shouldn't we put `done()` inside [then](https://github.com/EventStore/EventStore-Client-NodeJS/pull/209/files#diff-2eb2523855828faaaf5ed34a589b45aae8ddd3b312d2d5cced0125a559e52235R30)? Having it here may create false/positive results if the catch wasn't called for some reason (e.g. timing issue).